immuneML/reports/encoding_reports/FeatureDistribution.py

```diff
@@ -1,35 +1,40 @@
-import warnings
 from pathlib import Path
 
 import plotly.express as px
 
-from immuneML.analysis.data_manipulation.DataReshaper import DataReshaper
-from immuneML.data_model.dataset.RepertoireDataset import RepertoireDataset
+from immuneML.data_model.dataset.Dataset import Dataset
 from immuneML.reports.ReportOutput import ReportOutput
-from immuneML.reports.ReportResult import ReportResult
-from immuneML.reports.encoding_reports.EncodingReport import EncodingReport
-from immuneML.util.PathBuilder import PathBuilder
+from immuneML.reports.encoding_reports.FeatureReport import FeatureReport
 
 
-
-class FeatureDistribution(EncodingReport):
+class FeatureDistribution(FeatureReport):
```

Comment:
if we do this, the class hierarchy could get too deep and it can be harder to understand the class behaviour. Do you think it could work as well if refactored to have the common functionality in a helper class that all these reports would call? might be cleaner that way :)

Reply:
I did think about this, I chose to do it with inheritance because it is only the 'plot' method that is different so with a helper class it wouldnt be possible to share the init or _generate method (which relies on a lot of 'self' calls). But if you have a strong preference for a helper class I can refactor it, it would be a bit more repetitive code that way

Reply:
Ok :) since nothing else depends on these reports, I guess this is fine. There might be some refactoring to be done there at some point (why is the inheritance easier in the first place - shared state etc), but this is ok for now :) 👍